### PR TITLE
Added reference to Spacemacs in the Emacs section

### PR DESCRIPTION
**Pull request by addisonamiri**

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ customization safe and easy.
 * [Cask](https://github.com/cask/cask) is a package manager for Emacs.
 * [Prelude](https://github.com/bbatsov/prelude) is an enhanced Emacs 24 distribution that should make your experience with Emacs both more pleasant and more powerful.
 * [use-package](https://github.com/jwiegley/use-package) is a declaration macro for simplifying your `.emacs`
+* [Spacemacs](https://github.com/syl20bnr/spacemacs) is a Emacs 24 distribution that builds on Evil-mode with ports of popular Vim plugins to closer emulate a Vim environment.
```
Comment (unixorn) [inline index.md]: Please move this before use-package so the entries are in alphabetical order.

Thanks for the contribution!

**Comment by addisonamiri** [comment]
Put it in alphabetic order.

**Comment by unixorn** [comment]
Thanks for making the change for me, and for the contribution

**Comment by addisonamiri** [comment]
No problem :)

**Pull request merged**